remote-api/src/test/java/org/alfresco/rest/api/impl/CategoriesImplTest.java

```diff
@@ -366,12 +371,149 @@ public void testCreateCategories_nonExistingParentNode()
+        final List<ChildAssociationRef> childAssociationRefMocks = prepareChildAssocMocks(childrenCount, parentCategoryNodeRef);
+        given(nodeServiceMock.getChildAssocs(parentCategoryNodeRef)).willReturn(childAssociationRefMocks);
+        childAssociationRefMocks.forEach(this::prepareCategoryNodeMocks);
+
+        //when
+        final CollectionWithPagingInfo<Category> categoryChildren = objectUnderTest.getCategoryChildren(PARENT_ID, parametersMock);
+
+        then(nodesMock).should().validateNode(PARENT_ID);
+        then(nodesMock).should().isSubClass(parentCategoryNodeRef, ContentModel.TYPE_CATEGORY, false);
+        then(nodeServiceMock).should().getChildAssocs(parentCategoryNodeRef);
+        childAssociationRefMocks.forEach(ca -> {
+            then(nodesMock).should().getNode(ca.getChildRef().getId());
+            then(nodeServiceMock).should()
+                    .getChildAssocs(ca.getChildRef(), RegexQNamePattern.MATCH_ALL, RegexQNamePattern.MATCH_ALL, false);
+            then(nodeServiceMock).should().getPrimaryParent(ca.getChildRef());
+        });
+        then(nodeServiceMock).should(times(childrenCount)).getParentAssocs(parentCategoryNodeRef);
+
+        then(nodesMock).shouldHaveNoMoreInteractions();
+        then(nodeServiceMock).shouldHaveNoMoreInteractions();
+
+        then(authorityServiceMock).shouldHaveNoInteractions();
+        then(categoryServiceMock).shouldHaveNoInteractions();
+
+        assertEquals(childAssociationRefMocks.size(), categoryChildren.getTotalItems().intValue());
+        final List<Category> categoryChildrenList = new ArrayList<>(categoryChildren.getCollection());
+        IntStream.range(0, childrenCount).forEach(i -> assertCategoryFields(categoryChildrenList.get(i), i, PARENT_ID));
+    }
+
+    @Test
+    public void testGetCategoryChildren_noChildren()
+    {
+        final NodeRef parentCategoryNodeRef = new NodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE, PARENT_ID);
+        given(nodesMock.validateNode(PARENT_ID)).willReturn(parentCategoryNodeRef);
+        given(nodesMock.isSubClass(parentCategoryNodeRef, ContentModel.TYPE_CATEGORY, false)).willReturn(true);
+
+        given(nodeServiceMock.getChildAssocs(parentCategoryNodeRef)).willReturn(Collections.emptyList());
+
+
+        //when
+        final CollectionWithPagingInfo<Category> categoryChildren = objectUnderTest.getCategoryChildren(PARENT_ID, parametersMock);
+
+        then(nodesMock).should().validateNode(PARENT_ID);
+        then(nodesMock).should().isSubClass(parentCategoryNodeRef, ContentModel.TYPE_CATEGORY, false);
+        then(nodesMock).shouldHaveNoMoreInteractions();
+        then(nodeServiceMock).should().getChildAssocs(parentCategoryNodeRef);
+        then(nodeServiceMock).shouldHaveNoMoreInteractions();
+
+        then(authorityServiceMock).shouldHaveNoInteractions();
+        then(categoryServiceMock).shouldHaveNoInteractions();
+
+        assertEquals(0, categoryChildren.getTotalItems().intValue());
```

Comment:
Is it worth also checking the actual list is empty? It looks like the `totalItems` is pretty independent of the `collection`.

Reply:
Good point - adding that.